### ISSUE TITLE
URD -> destiny -> VIM -> car only available for R3.5

### ISSUE THREAD

**Issue by maximilianh** [farrellja/URD#48]
Hi, when running your installation script I'm getting an error, since the package "car" required by VIM required by destiny is not available for R < 3.5:

```
ERROR: dependency ‘VIM’ is not available for package ‘destiny’
* removing ‘/cluster/home/max/miniconda3/envs/r34/lib/R/library/destiny’

The downloaded source packages are in
        ‘/scratch/tmp/RtmpCX3hgU/downloaded_packages’
Updating HTML index of packages in '.Library'
Making 'packages.html' ... done
Error in eval(ei, envir) : 
  Failed to install required package 'destiny' from Bioconductor.
In addition: Warning messages:
1: In install.packages(pkgs = doing, lib = lib, ...) :
  installation of package ‘VIM’ had non-zero exit status
2: In install.packages(pkgs = doing, lib = lib, ...) :
  installation of package ‘destiny’ had non-zero exit status
> 
> 
> install.packages("VIM")
Warning: dependency ‘car’ is not available
trying URL 'http://cloud.r-project.org/src/contrib/VIM_4.8.0.tar.gz'
Content type 'application/x-gzip' length 342022 bytes (334 KB)
==================================================
downloaded 334 KB

ERROR: dependency ‘car’ is not available for package ‘VIM’
* removing ‘/cluster/home/max/miniconda3/envs/r34/lib/R/library/VIM’

The downloaded source packages are in
        ‘/scratch/tmp/RtmpCX3hgU/downloaded_packages’
Updating HTML index of packages in '.Library'
Making 'packages.html' ... done
Warning message:
In install.packages("VIM") :
  installation of package ‘VIM’ had non-zero exit status
> install.packages("car")
Warning message:
package ‘car’ is not available (for R version 3.4.1) 
```

car indeed requires R3.5
https://cran.r-project.org/web/packages/car/index.html

are you running URD on R3.4 ?

**Comment by MichaelPeibo** [comment]
Hi @maximilianh 
My best installing experience is by devtools.
`library(devtools) install_github(repo="farrellja/URD")`

**Comment by farrellja** [comment]
@maximilianh 

I, indeed, run URD in R 3.4.1. My sessionInfo is below.
I have version 4.7.0 of **VIM** installed and version 2.1-6 of **car**.
Perhaps the requirements have changed for version 3 of car; you could try installing an older version manually.

-- Jeff 

-----------------------------------------------------------------------

R version 3.4.1 (2017-06-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] URD_1.1.0.9004 Matrix_1.2-15  ggplot2_3.1.0  Seurat_3.0.2  

loaded via a namespace (and not attached):
  [1] backports_1.1.2     Hmisc_4.1-0         RcppEigen_0.3.3.3.1 plyr_1.8.4          igraph_1.2.4        lazyeval_0.2.1     
  [7] sp_1.2-5            splines_3.4.1       listenv_0.7.0       digest_0.6.18       htmltools_0.3.6     viridis_0.5.1      
 [13] gdata_2.18.0        magrittr_1.5        checkmate_1.8.5     cluster_2.0.6       ROCR_1.0-7          globals_0.12.4     
 [19] gmodels_2.18.1      R.utils_2.6.0       xts_0.10-1          colorspace_1.4-1    ggrepel_0.8.0       dplyr_0.8.3        
 [25] crayon_1.3.4        jsonlite_1.5        lme4_1.1-15         survival_2.41-3     zoo_1.8-0           ape_5.0            
 [31] glue_1.3.1          polyclip_1.10-0     gtable_0.2.0        MatrixModels_0.4-1  car_2.1-6           future.apply_1.3.0 
 [37] BiocGenerics_0.24.0 DEoptimR_1.0-8      SparseM_1.77        VIM_4.7.0           scales_1.0.0        bibtex_0.4.2       
 [43] Rcpp_1.0.2          metap_1.1           viridisLite_0.3.0   laeken_0.4.6        htmlTable_1.11.1    reticulate_1.10    
 [49] foreign_0.8-69      rsvd_1.0.1          proxy_0.4-20        SDMTools_1.1-221    Formula_1.2-3       tsne_0.1-3         
 [55] vcd_1.4-4           htmlwidgets_1.3     httr_1.3.1          gplots_3.0.1.1      RColorBrewer_1.1-2  acepack_1.4.1      
 [61] ica_1.0-1           pkgconfig_2.0.2     R.methodsS3_1.7.1   nnet_7.3-12         labeling_0.3        tidyselect_0.2.5   
 [67] rlang_0.4.0         reshape2_1.4.3      munsell_0.5.0       tools_3.4.1         ggridges_0.4.1      evaluate_0.10.1    
 [73] stringr_1.4.0       yaml_2.1.16         knitr_1.18          fitdistrplus_1.0-9  robustbase_0.93-1.1 caTools_1.17.1.2   
 [79] purrr_0.2.5         RANN_2.6.1          ggraph_1.0.2        pbapply_1.3-3       future_1.14.0       nlme_3.1-131       
 [85] quantreg_5.34       R.oo_1.21.0         compiler_3.4.1      pbkrtest_0.4-7      rstudioapi_0.7      plotly_4.8.0       
 [91] curl_3.1            png_0.1-7           e1071_1.7-1         smoother_1.1        tibble_2.1.3        tweenr_0.1.5       
 [97] stringi_1.4.3       lattice_0.20-35     nloptr_1.0.4        pillar_1.4.2        Rdpack_0.10-1       lmtest_0.9-35      
[103] data.table_1.12.0   cowplot_0.9.2       bitops_1.0-6        irlba_2.3.1         gbRd_0.4-11         R6_2.2.2           
[109] latticeExtra_0.6-28 KernSmooth_2.23-15  gridExtra_2.3       codetools_0.2-15    boot_1.3-20         MASS_7.3-48        
[115] gtools_3.8.1        assertthat_0.2.1    destiny_2.6.2       rprojroot_1.3-2     minpack.lm_1.2-1    withr_2.1.2        
[121] sctransform_0.2.0   diptest_0.75-7      mgcv_1.8-22         parallel_3.4.1      grid_3.4.1          rpart_4.1-11       
[127] tidyr_0.7.2         class_7.3-14        minqa_1.2.4         rmarkdown_1.8       Rtsne_0.15          TTR_0.23-2         
[133] ggforce_0.3.1       Biobase_2.38.0      base64enc_0.1-3

**Comment by farrellja** [comment]
@decarlin I think I've fixed this in the debug version, but was nervous about pushing it to the main branch without testing it more. (I've only used it once, but it let me install successfully with R 3.6.2) Try using the installation script from the debug branch, but following the installation instructions otherwise, and let me know if it works for you:
`source("https://raw.githubusercontent.com/farrellja/URD/debug/URD-Install.R")`

**Comment by tatarsky** [comment]
I wanted to post my notes from my attempt to load the above in a clean Anaconda R copy in case it helps others. OS in question is Ubuntu 16LTS. I am a sysadmin working with a person above. 

Installed Anaconda from scratch as a control. You could probably skip that step. 

`conda create -n r_env r-essentials r-base r-devtools`

Run R in that env. Which at this moment is a version 3.6.1.

`source("https://raw.githubusercontent.com/farrellja/URD/debug/URD-Install.R")`

When asked to update some existing packages I declined the first time but later hit a snag where the `cli` module was too old. So you may wish to do so. I may have fixed it by loading r-devtools in conda instead of the script. The error I noted was `Error: object 'cli_alert_info' is not exported by 'namespace:cli'`

When you get to this part of the script I got this error.

`devtools::install_github(repo="farrellja/URD")
Downloading GitHub repo farrellja/URD@master
sh: 1: /bin/gtar: not found
sh: 1: /bin/gtar: not found`

This appears to be a known conda issue and I hacked around it short term with a symlink from /bin/tar to /bin/gtar based on one of the hits for it (https://github.com/conda-forge/r-devtools-feedstock/issues/4)

It then proceeded when re-run manually to build. I don't recall any other issues 

> DONE (URD)
> library("URD")
> Loading required package: ggplot2
> Registered S3 methods overwritten by 'ggplot2':
  method         from 
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
Loading required package: Matrix
Registered S3 method overwritten by 'xts':
  method     from
  as.zoo.xts zoo

**Comment by decarlin** [comment]
Thanks, that seems to work with the up to date R!

**Comment by farrellja** [comment]
@tatarsky Thank you so much for this very clear description of your experience. It is extremely helpful to me. Looks like it's mostly ready to go, since it seems that most issues you had are related to conda. Do you agree?

**Comment by tatarsky** [comment]
Combined with the thumbs up from @decarlin who is the scientist (I am just the sysadmin) I would definitely agree! I believe even some of the Conda issues are actually fixed. I'll review that after I clear some other items!

**Comment by farrellja** [comment]
@tatarsky @decarlin Great! As a heads up, there is still an incompatibility between URD and ggraph 2.0.0+. I would recommend downgrading to 1.0.2 right now until I figure out what's going on. I forgot to change it in the package description, so it probably installed the newest version.

**Comment by farrellja** [comment]
Pushed version 1.1.1 to 'master' which should fix installation issues on R 3.5 and R 3.6, as well as the incompatibility with ggraph 2.0.0
